include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7533,6 +7533,13 @@ def err_nsconsumed_attribute_mismatch : Error<
 def err_nsreturns_retained_attribute_mismatch : Error<
   "overriding method has mismatched ns_returns_%select{not_retained|retained}0"
   " attributes">;
+
+def err_nserrordomain_not_tagdecl : Error<
+  "ns_error_domain attribute only valid on enum/struct/union/class">;
+def err_nserrordomain_invalid_decl : Error<
+  "domain argument %0 not valid top-level declaration">;
```

Comment:
How about "domain argument %0 does not refer to a global constant"? Even if we don't actually check all of those things, it guides the user in the right direction.